types/src/shared/cache.ts

```diff
@@ -49,3 +62,34 @@ export function cacheWithRedis<T extends (...args: any[]) => Promise<any>>(
     }
   };
 }
+
+const locks: Record<string, (() => void)[]> = {};
```

Comment:
I saw it the other way round : since we don't need this lock across deployments (and arguably we even want it to be cleared across deployments) why use redis which is more costly (network calls, etc)

Reply:
🤔 
For the current issue we are facing, a stateful lock works well. However, if we aim for a more resilient approach, a distributed lock might be the better option. Given that you eliminated the possibility of deadlocks, I'm comfortable for both solutions.